go.mod

```diff
@@ -3,8 +3,17 @@ module github.com/ipfs/go-path
 go 1.14
 
 require (
-	github.com/ipfs/go-cid v0.0.2
-	github.com/ipfs/go-ipld-format v0.0.2
-	github.com/ipfs/go-log v0.0.1
-	github.com/ipfs/go-merkledag v0.0.6
+	github.com/ipfs/go-blockservice v0.1.4
+	github.com/ipfs/go-cid v0.0.7
+	github.com/ipfs/go-datastore v0.4.5
+	github.com/ipfs/go-fetcher v1.2.0
+	github.com/ipfs/go-ipfs-blockstore v0.1.4
+	github.com/ipfs/go-ipfs-exchange-offline v0.0.1
+	github.com/ipfs/go-ipld-cbor v0.0.3
+	github.com/ipfs/go-ipld-format v0.2.0
+	github.com/ipfs/go-log v1.0.4
+	github.com/ipfs/go-merkledag v0.3.2
+	github.com/ipfs/go-unixfsnode v1.1.1
+	github.com/ipld/go-ipld-prime v0.9.1-0.20210402181957-7406578571d1
+	github.com/stretchr/testify v1.7.0
```

Comment:
It kind of bothers me that we've ended up with so many extra dependencies here, even though they're mostly just used for testing. I feel like @warpfork may have some thoughts/suggestions here.

Reply:
Yeah... there's a mixture of things going on here though...

- I bet a couple of these are new because the `go mod tidy` tool is listing more things lately (and they're not actually new dependencies).
- Yeah, I'd like this list to be less...
- ... but testing is a predominatingly worthy cause, if that's what's causing them to appear.
- It's maybe not worth spending a ton of time on it in this repo, if we hope to erode this repo into the abyss entirely.  (But let's watchdog intensely on the things that replace it, yeah.)
- The real problem, no matter which way we slice it, is... if you really wanna be sad, go look at the lock file.  Somehow we've got transitive dependencies sprawling all the way to `etcd` and `btcd` (?!).  I don't know which of these direct dependencies has *those* wildly out of control transitive sprawls, but that's probably the thing I'd worry about first.

Reply:
There a few less now. Some are just new dependencies (the fetcher). A few are yes -- for testing. While we could drop a bunch of these by switching away from protonodes, I'd rather not we ship v0.10.0 with these changes -- the main consume of this functionality is UnixFS, so it feels important to test the main exercised code path.